tests/integration/test_charm.py

```diff
@@ -20,20 +26,63 @@
 class TestDeployment:
     """Integration tests for charm."""
 
-    async def test_livepatch_server(self, ops_test: OpsTest):
-        """Perform GET request on the Livepatch unit."""
-        url = await get_unit_url(ops_test, application=APP_NAME, unit=0, port=8080)
-        logger.info("curling app address: %s", url)
-        response = requests.get(url, timeout=300, verify=False)  # nosec
+    async def assert_http(
+        self, ops_test: OpsTest, protocol: str, application: str, unit: int, port: int, path: str = ""
+    ):
+        """Call given HTTP endpoint and assert if the call was successful (HTTP 200)"""
+        u = await get_unit_url(ops_test, protocol=protocol, application=application, unit=unit, port=port)
+        u += ("/" + path) if path else ""
```

Comment:
Nit: I think we can use Python's [urlparse](https://stackoverflow.com/questions/8223939/how-to-join-absolute-and-relative-urls) package here.

Reply:
I'll apply this in the follow-up PR (#35).